signer/src/transaction_coordinator.rs

```diff
@@ -305,6 +316,28 @@ where
             };
         }
 
+        // If we have just deployed the smart contracts, trying to broadcast the
+        // key rotation tx would result in transaction rejected, reason: NoSuchContract
+        if self.sbtc_contracts_deployed && !just_deployed_contracts {
```

Comment:
suggestion: i personally might have put this into a new function just to keep things concise (i probably would have done the same for the actual deployment above too but that was another PR)

Reply:
Refactored a bit in [0febb85](https://github.com/stacks-network/sbtc/pull/788/commits/0febb856000e877689aac10d7cb00002b1eff071)